src/Model/Filter/Like.php

```diff
@@ -78,11 +85,33 @@ public function process()
         }
 
         if (!empty($conditions)) {
-            $this->query()->andWhere([$this->config('fieldMode') => $conditions]);
+            $this->query()->andWhere([$this->config('fieldMode') => $conditions], $colTypes);
         }
     }
 
     /**
+     * @param string $field Field
+     * @return bool
+     */
+    protected function _requiresTypeMapping($field)
+    {
+        if (strpos($field, '.') === false) {
+            return false;
+        }
+        list($model, $field) = explode('.', $field);
```

Comment:
This could still find the wrong model here, right? I dont quite like it yet.
The actual base class should be doing this prior to aliasing the field probably.
Any ideas here?